integration-tests/bulkExportTestHelper.ts

```diff
@@ -97,6 +116,36 @@ export default class BulkExportTestHelper {
         }
     }
 
+    async createGroupAndReturnResources(groupMemberMeta?: GroupMemberMeta) {
+        try {
+            const response = await this.fhirUserAxios.post('/', createGroupMembersBundle);
+            const createGroupBundle = cloneDeep(createGroupAndPatientCompartmentBundle);
+            // Update group members to resources just created
+            createGroupBundle.entry[1].resource.member = response.data.entry.map(
+                (item: { response: { location: any } }) => ({
+                    entity: { reference: item.response.location },
+                    ...groupMemberMeta,
+                }),
+            );
+            // Update reference in Encounter to reference a patient just created
```

Comment:
I think that the bundle manipulation done to set up the test data can be greatly simplified.

Looks like all the data (with proper reference set between resources) for your test scenarios can be created in a single bundle transaction if you use internal references within the bundle. 

See: 
- https://www.hl7.org/fhir/bundle.html#references
- https://www.hl7.org/fhir/bundle-references.json.html

On the example json, the important part is that there's an `Observation` with:
```
"subject": {
          "reference": "urn:uuid:04121321-4af5-424c-a0e1-ed3aab1c349d"
        }
```

and a `Patient` with:
```
{
      "fullUrl": "urn:uuid:04121321-4af5-424c-a0e1-ed3aab1c349d",
      "resource": {
        "resourceType": "Patient",
```

---
An alternative would be to just create resources one at a time without using bundles. There's already a `randomPatient` method, and building a `Group` resource is easy. Although you probably want to do the Bundle thing since your code is already invested in using Bundles.